asyncpgx/connection.py

```diff
@@ -27,6 +27,25 @@ def _prepare_asyncpg_parameters(
     async def named_execute(self, query: str, args: typing.Dict, timeout: typing.Optional[float] = None) -> str:
         """Extended versions of `execute` with support of the named
         parameters."""
+        # Convert the received query in the parameter to all lower-case
```

Comment:
Hi!
Thank you for PR! I'd like to suggest a couple improvements
1. we need this validation on every DB-access method not just `named_execute` (i.e. `named_executemany`, `named_fetch`, etc.). So i suggest to move this logic to `asyncpgx.query` module. As i can see the best option will be implement this logic in base `QueryParamsConverter` class and call it in child classes. 
2. another problem - we need validation logic for every type of query, not just `INSERT INTO` one. This problem also could be fixed by moving logic to `QueryParamsConverter` class, because it already gets all arguments so there is no need to parse the query. All we need to do is compare `original_args` and `params_order_list` parameters.
3. I'd prefer specific exception instead of KeyError, something like
```
class NamedParametersValidationError(Exception):
    pass
```
4. For now this project is fully covered by tests, and i'd be very pleased if you'll write a couple of tests for this functionality. Taking into account all i've wrote before, the best option would be a couple of unit-tests in `asyncpgx/tests/test_query.py` module which will fully cover the new functionality.

Reply:
Much thanks in advance!

Reply:
I will take a stab, as time permits. Thank you for your feedback!